src/reformat/rgb.rs

```diff
@@ -319,6 +310,101 @@ impl Image {
         }
         Ok(())
     }
+
+    pub fn view(&self) -> Image {
+        Image {
+            width: self.width,
+            height: self.height,
+            depth: self.depth,
+            format: self.format,
+            chroma_upsampling: self.chroma_upsampling,
+            chroma_downsampling: self.chroma_downsampling,
+            alpha_premultiplied: self.alpha_premultiplied,
+            is_float: self.is_float,
+            max_threads: self.max_threads,
+            pixels: match &self.pixels {
+                Some(Pixels::Pointer(ptr)) => Some(Pixels::Pointer(*ptr)),
+                Some(Pixels::Buffer(vec)) => Some(Pixels::Pointer(vec.as_ptr() as *mut u8)),
+                Some(Pixels::Buffer16(vec)) => {
+                    Some(Pixels::Pointer(vec.as_ptr().cast::<u8>() as *mut u8))
+                }
+                None => None,
+            },
+            row_bytes: self.row_bytes,
+        }
+    }
+
+    pub fn shuffle_channels_to(self, format: Format) -> AvifResult<Image> {
+        if self.format == format {
+            return Ok(self);
+        }
+        if self.format == Format::Rgb565 || format == Format::Rgb565 {
+            return Err(AvifError::NotImplemented);
+        }
+
+        let mut dst = Image {
```

Comment:
sorry i missed this earlier.

you can use the struct update syntax for this initialization (i.e.) override only the fields that you want and use `..self` in the end.

```rust
let mut dst = Image {
  format,
  pixels: None,
  row_bytes: 0,
  ..self
};
```

there might be other places where we could have done this (like line 57), but we can leave that for a separate PR.